getting_started/scripting/gdscript/static_typing.rst

```diff
@@ -0,0 +1,413 @@
+lead to runtime errors.
+
+You can configure warnings in the Project Settings under a new section
+called ``GDscript``:
+
+.. figure:: ./img/typed_gdscript_warning_system_settings.png
+   :alt: warning system project settings
+
+   warning system project settings
+
+You can find a list of warnings for the active GDscript file in the
+script editor’s status bar. The example below has 3 warnings:
+
+.. figure:: ./img/typed_gdscript_warning_example.png
+   :alt: warning system example
+
+   warning system example
+
+To ignore specific warnings in one file, insert a special comment of the
+form ``#warning-ignore:warning-id``, or click on the ignore link to the
+right of the warning’s description. Godot will add a comment above the
+corresponding line and the code won’t trigger the corresponding warning
+anymore:
+
+.. figure:: ./img/typed_gdscript_warning_system_ignore.png
+   :alt: warning system ignore example
+
+   warning system ignore example
+
+Warnings won’t prevent the game from running, but you can turn them into
+errors if you’d like. This way your game won’t compile unless you fix
+all warnings. Head to ``GDscript`` section of the Project Settings to
+turn on this option. Here’s the same file as the previous example with
+warnings as errors turned on:
+
+.. figure:: ./img/typed_gdscript_warning_system_errors.png
+   :alt: warnings as errors
+
+   warnings as errors
+
+Cases Where You Can’t Specify Types
+-----------------------------------
+
+To wrap up this introduction, let’s cover a few cases where you can’t
+use type hints. All the examples below **will trigger errors**.
+
+You can’t use Enums as types:
+
+::
+
+    enum MoveDirection { UP, DOWN, LEFT, RIGHT }
+    var current_direction : MoveDirection
+
+You can’t specify the type of individual members in an array. This will
+give you an error:
+
+::
+
+    var enemies : Array = [$Goblin : Enemy, $Zombie : Enemy]
+
+You can’t force the assignment of types in a ``for`` loop as each
+element the ``for`` keyword loops already has a different type. So you
+**cannot** write:
+
+::
+
+    var names ['John', 'Marta', 'Samantha', 'Jimmy']
+    for name : String in names:
+        pass
+
+Two scripts can’t depend on each other in a cyclic fashion:
+
+::
+
+    """Player.gd"""
+    extends Area2D
+    class_name Player
+
+    var rifle : Rifle
+
+::
+
+    """Rifle.gd"""
+    extends Area2D
+    class_name Rifle
+
+    var player : Player
+
+Summary
+-------
+
+Typed GDscript is a powerful tool. Coming with Godot 3.1, it can already
+help you write more structured code, help you avoid common errors, and
+create scalable systems. In the future, static types will also bring you
+a nice performance boost thanks to upcoming compiler optimizations.
+
+Want to become a better game creator? Be sure to `follow us on
+Twitter <https://twitter.com/NathanGDquest>`__ for more tutorials and
+news dedicated to game creation and Free Software!
```

Comment:
"Removed the link to my twitter at the end", you say? I think your Twitter snuck back in. :P